packages/gel/src/options.ts

```diff
@@ -46,18 +46,17 @@ export interface SimpleRetryOptions {
 export type WarningHandler = (warnings: errors.GelError[]) => void;
 
 export const throwWarnings: WarningHandler = (warnings) => {
-  throw new AggregateError(
-    warnings,
-    `Warnings occurred while running query: ${warnings.map((warn) => warn.message)}`,
-  );
+  throw new AggregateError(warnings, formatWarnings(warnings));
 };
 
 export const logWarnings: WarningHandler = (warnings) => {
-  for (const warning of warnings) {
-    console.warn(new Error(`Gel warning: ${warning.message}`));
-  }
+  const merged = new Error(formatWarnings(warnings));
+  console.warn(Object.assign(merged, { name: "" }));
 };
 
+const formatWarnings = (warnings: errors.GelError[]) =>
+  `warnings occurred while running query:\n${warnings.map((warn) => warn.message).join("\n")}`;
```

Comment:
Just bump: https://github.com/geldata/gel-js/blob/34bca0b58f9676e16014644b25fbb6355e90a815/packages/gel/test/client.test.ts#L2220 to do two separate searches maybe?

```ts
/warnings occurred while running query/
/Test warning please ignore/
```

?

Reply:
Actually it appears this is just testing the `throwWarnings` function. maybe we should just call it directly?

Reply:
That's tested above

Reply:
I think I'd still rather keep this test as a full integration example rather than break out a separate unit test. There is still some value here to catch if we accidentally swallow errors in some change to the `warningHandler` handling in the future. I'd be fine to _add_ a unit test though, too, but that seems pointless when we have an integration test covering the unit.